src/devices/Board/I2cBusManager.cs

```diff
@@ -68,13 +68,8 @@ public I2cBusManager(Board board, int bus, int[]? pins, I2cBus busInstance)
         /// <remarks>No test is performed whether the given device exists and is usable</remarks>
         public override I2cDevice CreateDevice(int deviceAddress)
         {
-            if (_devices.TryGetValue(deviceAddress, out I2cDevice? device))
-            {
-                return device;
-            }
-
             I2cDevice newDevice = _busInstance.CreateDevice(deviceAddress);
-            _devices.Add(deviceAddress, newDevice);
+            _devices[deviceAddress] = newDevice;
```

Comment:
at this point we should consider making `_devices` either `HashSet<I2cDevice>` or `Dictionary<int, HashSet<I2cDevice>>` - perhaps even change design so that `RemoveDevice(int)` is replaced with `RemoveDevice(I2cDevice)`. Alternatively we should throw on CreateDevice if it was created again (we should also check if it was already disposed though). My personal favor is we should change the design.

@pgrawehr thoughts?

Reply:
That would probably be better, but I'm not sure it's worth the effort. Would need to make the changes and then check whether they show any benefit.

Reply:
let's do it separately if needed in that case - current state is still better than it was - that's why I approved in the first place